universe/Building.h

```diff
@@ -214,9 +214,8 @@ class BuildingTypeManager {
 
 private:
     BuildingTypeManager();
-    ~BuildingTypeManager();
 
-    std::map<std::string, BuildingType*> m_building_types;
+    std::map<std::string, std::unique_ptr<BuildingType>> m_building_types;
```

Comment:
> Yes it won't be deleted, but it also can't be safely dereferenced in either thread.

The `shared_ptr` can never be safely dereferenced, or the member functions of the pointed-to-object may not be threadsafe, depending on how they are implemented?